profile/src/main/java/com/scottlogic/deg/profile/reader/ConstraintReader.java

```diff
@@ -0,0 +1,220 @@
+        return constraints.stream()
+                .map(subConstraintDto -> read(subConstraintDto, fields))
+                .collect(Collectors.toSet());
+    }
+
+    public Constraint read(ConstraintDTO dto, ProfileFields profileFields)
+    {
+        if (dto == null) throw new InvalidProfileException("Constraint is null");
+        if (dto instanceof RelationalConstraintDTO) return readRelationalConstraintDto((RelationalConstraintDTO)dto, profileFields);
+        return dto instanceof AtomicConstraintDTO
+                ? readAtomicConstraintDto((AtomicConstraintDTO) dto, profileFields)
+                : readGrammaticalConstraintDto(dto, profileFields);
+    }
+
+    private Constraint readRelationalConstraintDto(RelationalConstraintDTO dto, ProfileFields fields)
+    {
+        Field main = fields.getByName(dto.field);
+        Field other = fields.getByName(dto.getOtherField());
+        switch (dto.getType())
+        {
+            case EQUAL_TO_FIELD:
+                Granularity offsetGranularity = readGranularity(main.getType(), dto.offsetUnit);
+                return offsetGranularity != null
+                        ? new EqualToOffsetRelation(main, other, offsetGranularity, dto.offset)
+                        : new EqualToRelation(main, other);
+            case AFTER_FIELD:
+                return new AfterRelation(main, other, false, DateTimeDefaults.get());
+            case AFTER_OR_AT_FIELD:
+                return new AfterRelation(main, other, true, DateTimeDefaults.get());
+            case BEFORE_FIELD:
+                return new BeforeRelation(main, other, false, DateTimeDefaults.get());
+            case BEFORE_OR_AT_FIELD:
+                return new BeforeRelation(main, other, true, DateTimeDefaults.get());
+            case GREATER_THAN_FIELD:
+                return new AfterRelation(main, other, false, NumericDefaults.get());
+            case GREATER_THAN_OR_EQUAL_TO_FIELD:
+                return new AfterRelation(main, other, true, NumericDefaults.get());
+            case LESS_THAN_FIELD:
+                return new BeforeRelation(main, other, false, NumericDefaults.get());
+            case LESS_THAN_OR_EQUAL_TO_FIELD:
+                return new BeforeRelation(main, other, true, NumericDefaults.get());
+            default:
+                throw new InvalidProfileException("Unexpected relation data type " + dto.getType());
+        }
+    }
+
+    private Constraint readAtomicConstraintDto(AtomicConstraintDTO dto, ProfileFields profileFields)
+    {
+        Field field = profileFields.getByName(dto.field);
+        switch (dto.getType())
+        {
+            case EQUAL_TO:
+                return new EqualToConstraint(field, readAnyType(field, ((EqualToConstraintDTO)dto).value));
+            case IN_SET:
+                InSetConstraintDTO inSetConstraintDTO = (InSetConstraintDTO) dto;
+                DistributedList<Object> values = (inSetConstraintDTO instanceof InSetFromFileConstraintDTO
```

Comment:
Could probably do with moving this logic out of the switch, for readablility

Reply:
Will add this to #1461